src/form.js

```diff
@@ -308,6 +309,11 @@ define([
                         meta = null;
                     }
                     this.knownInstances[attrs.id] = attrs.src;
+                } else if (meta.attributes.id !== attrs.id) {
+                    // support renaming when there is no reference
+                    if (meta.internal) {
+                        meta.attributes.id = attrs.id;
+                    }
                 }
             } else if (attrs.id) {
                 // attrs has no src, find by id
```

Comment:
> We may also need to somehow track those instances in the knownInstances mapping.

The fact that we are not tracking data instances (without `src`) in `form.knownInstances` is definitely the problem. I think instead of having

`knownInstances = {<instance id>: <instance src>}`

we should have

`knownInstances = {<instance id>: <instance metadata>}`